src/Particle/HDFWalkerInput_0_4.cpp

```diff
@@ -168,11 +169,17 @@ bool HDFWalkerInput_0_4::read_hdf5(const std::filesystem::path& h5name)
     const int nitems    = num_ptcls_ * OHMMS_DIM;
     const int curWalker = wc_list_.getActiveWalkers();
     wc_list_.createWalkers(nw_in, num_ptcls_);
-    Buffer_t::iterator it(posin.begin() + woffsets[myComm->rank()] * nitems);
+    Buffer_t::iterator it = posin.begin() + woffsets[myComm->rank()] * nitems;
+    auto it_w             = weights_in.begin() + woffsets[myComm->rank()];
```

Comment:
Can we just compute it and it_w inside the loop? iterator is so hard to follow.

Reply:
I refactored those 3 loops so they're easier to read.